storage/ftp_storage.py

```diff
@@ -139,13 +139,19 @@ def save_to_file(self, out_file: BinaryIO) -> None:
         ftp_client = self._connect()
         filename = self._cd_to_file(ftp_client)
 
-        ftp_client.retrbinary("RETR {0}".format(filename), callback=out_file.write)
+        try:
+            ftp_client.retrbinary("RETR {0}".format(filename), callback=out_file.write)
+        except error_perm:
+            raise NotFoundError("No File Found")
 
     def save_to_directory(self, destination_directory: str) -> None:
         ftp_client = self._connect()
         base_ftp_path = self._parsed_storage_uri.path
 
-        ftp_client.cwd(base_ftp_path)
+        try:
+            ftp_client.cwd(base_ftp_path)
+        except error_perm:
+            raise NotFoundError("No Files Found")
 
         for root, dirs, files in self._walk(ftp_client):
```

Comment:
I believe the `_walk` method can also raise because it navigates the directory structure on the FTP server.

Should we wrap the whole `save_to_directory` method in a single `try`...`except`?